tools/maint/win32_launcher/launcher.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2025 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+ * Small win32 application that is used to launcher emscripten via python.exe.
+ * On non-windows platforms this is done via the run_pyton.sh shell script.
+ *
+ * The binary will look for a python script that matches its own name and run
+ * that using python.exe.
+ */
+
+// Define _WIN32_WINNT to Windows 7 for max portability
+#define _WIN32_WINNT 0x0601
+
+#include <windows.h>
+#include <shellapi.h>
+#include <shlwapi.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#pragma comment(lib, "shlwapi.lib")
+#pragma comment(lib, "shell32.lib")
+
+wchar_t* get_python_executable() {
+  wchar_t* python_exe_w = _wgetenv(L"EMSDK_PYTHON");
+  if (!python_exe_w) {
+    return L"python";
+  }
+  return python_exe_w;
+}
+
+wchar_t* get_long_path(const wchar_t* path) {
+  DWORD path_len = GetFullPathNameW(path, 0, NULL, NULL);
+  if (path_len == 0)
+    abort();
+
+  wchar_t* full_path = malloc(sizeof(wchar_t) * (path_len + 4));
+  if (!full_path)
+    abort();
+
+  wcscpy_s(full_path, path_len + 4, L"\\\\?\\");
+  if (GetFullPathNameW(path, path_len, full_path + 4, NULL) == 0)
+    abort();
+  return full_path;
+}
+
+// Get the name of the currently running executable (module)
+wchar_t* get_module_path() {
+  DWORD buffer_size = MAX_PATH;
+  wchar_t* module_path_w = malloc(sizeof(wchar_t) * buffer_size);
+  if (!module_path_w)
+    abort();
+
+  DWORD path_len = GetModuleFileNameW(NULL, module_path_w, buffer_size);
+  while (path_len > 0 && path_len == buffer_size) {
+    buffer_size *= 2;
+    module_path_w = realloc(module_path_w, sizeof(wchar_t) * buffer_size);
+    if (!module_path_w)
+      abort();
+    path_len = GetModuleFileNameW(NULL, module_path_w, buffer_size);
+  }
+
+  if (path_len == 0)
+    abort();
+
+  return module_path_w;
+}
+
+// Handle the _EMCC_CCACHE environment variable which essentially re-executes
```

Comment:
IIUC we don't need to re-execute the launcher itself, just append `ccache.exe` before building up the rest of python command.